samples/go/csv/csv-import/importer.go

```diff
@@ -63,7 +64,7 @@ func main() {
 	switch {
 	case flag.NArg() == 0:
 		err = errors.New("Maybe you put options after the dataset?")
-	case flag.NArg() == 1 && *path == "":
+	case flag.NArg() == 1 && *path == "" && !*detectColumnTypes:
```

Comment:
I'm trying to understand why this last " && !*detectColumnTypes". It doesn't seem like you want this check here.